odl/tomo/operators/spect_trafo.py

```diff
@@ -0,0 +1,331 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+"""Single-photon emission computerized tomography (SPECT) projectors """
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+try:
+    from NiftyPy.NiftyRec import (SPECT_project_parallelholes,
+                                  SPECT_backproject_parallelholes)
+    NIFTYREC_AVAILABLE = True
+except ImportError:
+    NIFTYREC_AVAILABLE = False
+
+import numpy as np
+
+from odl.operator import Operator
+from odl import uniform_discr_frompartition
+from odl.discr.lp_discr import DiscreteLp
+from odl.tomo.geometry.spect import ParallelHoleCollimatorGeometry
+
+
+__all__ = ('AttenuatedRayTransform', 'AttenuatedRayBackprojection',
+           'NIFTYREC_AVAILABLE')
+_SUPPORTED_IMPL = ('niftyrec', 'niftyrec_gpu')
+
+
+class AttenuatedRayTransform(Operator):
+
+    """The discrete attenuated Ray transform between L^p spaces."""
+
+    def __init__(self, dom, geometry, attenuation=None, psf=None,
+                 impl='niftyrec'):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        dom : `DiscreteLp`
+            Discretized space, a 3 dimensional domain of the forward projector
+        geometry : `ParallelHoleCollimatorGeometry`
+            The geometry of the transform. An ODL geometry instance
```

Comment:
Lots of redundancy here. "Geometry of the transform" should do, unless we need to add more information.